packages/thirdweb/src/wallets/in-app/core/wallet/in-app-core.ts

```diff
@@ -123,12 +129,15 @@ export function createInAppWallet(args: {
         const ecosystemOptions = await getEcosystemInfo(ecosystem.id);
         const smartAccountOptions = ecosystemOptions?.smartAccountOptions;
         if (smartAccountOptions) {
-          const preferredChain = options.chain;
-          if (!preferredChain) {
```

Comment:
Wait this will always throw if there's no value set, won't that break everyone?

Why don't you throw below, only if there's no default chain AND no chain passed in in code?